src/pages/index.js

```diff
@@ -1,75 +1,182 @@
-   renderer: (item) => {
-     const cardElement = createCard(item);
-     cardSection.addItem(cardElement);
-   }
- }, ".cards__list");
-
- cardSection.renderItems();
-
- // Enable validation for all forms
- enableValidation(validationSettings);
-
- // Event listeners for opening modals
- profileEditButton.addEventListener("click", () => {
-   const userInfoData = userInfo.getUserInfo();
-   popupWithFormProfile.setInputValues(userInfoData);
-   formValidators['profile-form'].resetValidation();
-   popupWithFormProfile.open();
- });
-
- addNewCardButton.addEventListener("click", () => {
-   popupWithFormAddCard.open();
- });
+import Api from "../components/Api.js";
+import { enableValidation, formValidators } from "../components/FormValidator.js";
+import { validationSettings } from "../utils/Constants.js";
+
+// Elements
+const profileEditButton = document.querySelector("#profile-edit-button");
+const addNewCardButton = document.querySelector(".profile__add-button");
+const avatarEditButton = document.querySelector(".profile__image-edit-button");
+
+// API Instance
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "703aeebf-8509-4c4b-9691-d3d6d590ee64",
+    "Content-Type": "application/json"
+  }
+});
+
+// UserInfo instance
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image"
+});
+
+// Popup instances
+const popupWithImage = new PopupWithImage("#modal-preview-picture");
+popupWithImage.setEventListeners();
+
+const popupWithFormProfile = new PopupWithForm("#profile-edit-modal", handleProfileFormSubmit);
+popupWithFormProfile.setEventListeners();
+
+const popupWithFormAddCard = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+popupWithFormAddCard.setEventListeners();
+
+const popupWithFormAvatar = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+popupWithFormAvatar.setEventListeners();
+
+const popupWithConfirmation = new PopupWithConfirmation({
+  popupSelector: '#delete-confirmation-modal',
+  handleConfirm: handleDeleteCard
+});
+popupWithConfirmation.setEventListeners();
+
+// Card section instance
+const cardSection = new Section(
+  {
+    renderer: (cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+    }
+  },
+  ".cards__list"
+);
+
+// Load user information and initial cards
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, initialCards]) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error("Error loading initial data:", err);
+  });
+
+// Handle profile form submission
+function handleProfileFormSubmit(formData) {
+  popupWithFormProfile.renderLoading(true); 
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```